src/service/Fetch.ts

```diff
@@ -18,47 +18,89 @@ export type Configuration = {
   headers?: { [header: string]: string }
 }
 
-async function _fetch<ResultType>(
-  method: string,
-  route: string,
-  configuration?: Configuration,
-  body?: any
-): Promise<ResultType> {
-  if (!configuration) throw new Error("Cannot make HTTP request due to invalid configuration.")
-  let protocol = configuration!.base.includes("api") ? "https" : "http" // TODO: Remove after we move to https only
-  var result = await (
-    await fetch(`${protocol}://${configuration!.base}${route}`, {
-      method: method,
-      headers: new Headers({
-        "Content-Type": "application/json",
-        Accept: "application/json",
-        ...(configuration!.headers || {}),
-        Authorization: !!configuration!.authorization ? `${configuration!.authorization}` : undefined
-      } as any),
-      body: body !== undefined ? JSON.stringify(body) : undefined
-    })
-  ).json()
-  
-
-  //if (result['error'] !== undefined)
-  //    throw new Error(result['error'])
-  return result as any
+export type Auth = {
+  accessToken: string,
+  refreshToken: string,
 }
 
 export class Fetch {
+  private static auth?: Auth
+  public static update_auth(auth: Auth) {
+    Fetch.auth = auth
+  }
+
+  private static async _fetch<ResultType>(
+    method: string,
+    route: string,
+    configuration?: Configuration,
+    body?: any
+  ): Promise<ResultType> {
+    if (!configuration) throw new Error("Cannot make HTTP request due to invalid configuration.")
+    let protocol = configuration!.base.includes("api") ? "https" : "http" // TODO: Remove after we move to https only
+    var result = await (
+      await fetch(`${protocol}://${configuration!.base}${route}`, {
+        method: method,
+        headers: new Headers({
+          "Content-Type": "application/json",
+          Accept: "application/json",
+          ...(configuration!.headers || {}),
+          Authorization: !!Fetch.auth?.accessToken ? `Bearer ${Fetch.auth.accessToken}` : undefined
+        } as any),
+        body: body !== undefined ? JSON.stringify(body) : undefined
+      })
+    ).json()
+  
+    if (result.error === "403.invalid-token") { // get new token
```

Comment:
We should not use error messages, but response status as an indicator. We should check for status `401 Unauthorized` to see if we should attempt a token refresh.